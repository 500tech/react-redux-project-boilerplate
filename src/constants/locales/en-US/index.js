// @flow
import type { Locale } from 'types/common.types';
import translations from './en-US.translations.json';

const enUS: Locale = {
  dateTimeFormat: {
    parentLocale: 'en',
    longDateFormat: {
      LT: 'HH:mm',
      L: 'MM/DD/YYYY',
      LL: 'MM/DD/YYYY HH:mm'
    },
    week: {
      dow: 1
    }
  },
  translations
};

export default enUS;
