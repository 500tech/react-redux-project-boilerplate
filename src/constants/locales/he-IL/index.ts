import { Locale } from 'types/common.types';
import translations from './he-IL.translations.json';

const heIL: Locale = {
  dateTimeFormat: {
    parentLocale: 'he',
    longDateFormat: {
      LT: 'HH:mm',
      L: 'DD/MM/YYYY',
      LL: 'DD/MM/YYYY HH:mm'
    },
    week: {
      dow: 1
    }
  },
  translations
};

export default heIL;
