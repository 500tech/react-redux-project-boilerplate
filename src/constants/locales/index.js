// @flow
import enUS from 'constants/locales/en-US';

export default {
  'en-US': enUS
};
