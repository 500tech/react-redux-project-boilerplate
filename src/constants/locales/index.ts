import enUS from 'constants/locales/en-US';
import heIL from 'constants/locales/he-IL';
import { Locale } from 'types/common.types';

export type LocaleTypes = 'en-US' | 'he-IL';

interface LocaleList {
  [key: string]: Locale;
}

const locales: LocaleList = {
  'en-US': enUS,
  'he-IL': heIL
};

export default locales;
