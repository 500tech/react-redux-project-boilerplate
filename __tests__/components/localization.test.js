import React from 'react';
import { shallow } from 'enzyme';
import { noop, indexOf } from 'lodash/fp';

import locales from 'constants/locales';
import moment from 'moment/moment';
import forEach from 'lodash/forEach';

const setup = ({ locale = 'en-US' } = {}) => ({
  locale
});

describe.only('<Localization />', () => {
  let momentSpy, storeSpy, Localization, subscribeCallback;

  beforeEach(() => {
    momentSpy = {
      defineLocale: jest.fn(),
      locale: jest.fn()
    };

    storeSpy = {
      subscribe: jest.fn(callback => {
        subscribeCallback = callback;
      }),
      getState: jest.fn(() => ({ localization: { locale: 'en-US' } }))
    };

    jest.doMock('moment', () => momentSpy);
    jest.doMock('store', () => storeSpy);

    Localization = require('components/localization').Localization;
  });

  afterEach(() => {
    jest.resetModules();
  });

  test('should render', () => {
    const props = setup();
    const component = shallow(
      <Localization {...props}>
        <div>Hello world</div>
      </Localization>
    );

    expect(component).toMatchSnapshot();
  });

  test('should define locales in moment', () => {
    let index = 0;
    forEach(locales, (locale, key) => {
      expect(momentSpy.defineLocale.mock.calls[index]).toEqual([
        key,
        locale.dateTimeFormat
      ]);
      index += 1;
    });
  });

  test('should set moment default locale from the store', () => {
    expect(momentSpy.locale).toHaveBeenCalledWith('en-US');
  });

  test('should set moment default locale when it changes in the store', () => {
    storeSpy.getState = jest.fn(() => ({ localization: { locale: 'he-IL' } }));
    subscribeCallback();

    expect(momentSpy.locale).toHaveBeenCalledWith('he-IL');
  });
});
