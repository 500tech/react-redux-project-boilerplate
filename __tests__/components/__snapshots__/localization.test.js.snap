// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Localization /> should render 1`] = `
<IntlProvider
  key="en-US"
  locale="en-US"
  messages={
    Object {
      "homepage.title": "Welcome to the 500Tech boilerplate",
    }
  }
>
  <div>
    Hello world
  </div>
</IntlProvider>
`;
